### PR TITLE
[MBL-1460] PPO view model analytics

### PR DESCRIPTION
**Pull request by stevestreza-ksr**
# 📲 What

This PR adds telemetry events for various PPO actions that the user might take.

# 🛠 How

New analytics events were added along with relevant types for PPO-specific analytics calls. This includes using the new ProjectAnalyticsFragment which the PPO GraphQL query was updated to use.

GraphQL data is now parsed into a new usage-specific model, PPOProjectCardModel. This lets us do validation on the card at initialization to make sure everything is there that is needed.

The PPOProjectCardViewModel, PPOViewModel, and all the related views were updated to use PPOProjectCardViewModel, and actions were updated to pass the relevant PPOProjectCardModel when buttons are tapped.

When those buttons are tapped, actions are triggered in the PPOViewModel which trigger Subjects. Those Subjects are listened to, and the appropriate analytics events are generated.

# ✅ Acceptance criteria

Tests should pass.

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/Tracking/KSRAnalytics.swift [hunk 1]
```diff
@@ -42,8 +42,10 @@ public final class KSRAnalytics {
     case messages // MessagesViewController
     case onboarding // CategorySelectionViewController, CuratedProjectsViewController
     case pledgeAddNewCard = "pledge_add_new_card"
+    case pledgeManager = "pledge_manager" // PPOView
```
Comment (ifosli) [inline Library/Tracking/KSRAnalytics.swift:45]: Nit: Maybe delete this unless you know you need it later? Also, out of curiosity, what part of PPO counts as a "pledgeManager" context? I would never have guessed based on the name that that'd be related to PPO.

**Review by ifosli — APPROVED** [review]
Looks great! I like how the different cards are templates now that can be used both in the previews and the tests!

**Review by scottkicks — APPROVED** [review]
Though i'll admit I'm a little lost on PPO nuances, this LGTM from a high level

File: Library/Tracking/KSRAnalytics.swift (hunk 1 above)
Comment (stevestreza-ksr) [reply Library/Tracking/KSRAnalytics.swift:45]: Will delete, this was for a backend event and got added by mistake.

**Pull request merged**